File: Civitas-Java/Civitas/src/main/java/civitas/SorpresaPORCASAHOTEL.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author elena
 */
public class SorpresaPORCASAHOTEL extends Sorpresa{
    
    SorpresaPORCASAHOTEL(int valor, String texto){   //Resto de sorpresas
        super(valor, texto);
    }
    
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        Jugador jugador = todos.get(actual);
        
        if(jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            jugador.modificarSaldo(valor*jugador.cantidadCasasHoteles());
        }
    }
    
    @Override
    public String toString(){
        return super.toString() + 
               "\nTipo Sorpresa: PORCASAHOTEL";
    }
}

File: Civitas-Ruby/lib/sorpresa_porcasahotel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative "diario"
require_relative "tipo_sorpresa"
require_relative "mazo_sorpresas"
require_relative "tablero"
require_relative "jugador"

module Civitas
  class SorpresaPorcasahotel < Sorpresa
    def initialize(valor, texto)
      super(valor, texto)
    end
    
    @Override
    def aplicarajugador(actual, todos)
      jugador = todos[actual]
      
      if jugadorcorrecto(actual, todos)
        informe(actual, todos)
        jugador.modificarsaldo(super.valor*jugador.cantidadcasashoteles)
      end
    end
    
    @Override
    def to_s
      super.to_s + '\nTipo Sorpresa: PORCASAHOTEL'
    end 
  end
end
